### ISSUE TITLE
297799165 Cannot upload file if it has not been processed (Dale)

### ISSUE THREAD

**Issue by FreeREGcomputer** [FreeUKGen/FreeUKRegProductIssues#1529]
Issue reported by **STSCC** at 2024-06-08 17:59:25 UTC
Time: 2024-06-08T17:51:19+00:00
Session ID: f538d25189a0a54430076ee7aa988742
Problem Page URL: [/csvfiles/new](/csvfiles/new)
Previous Page URL: [https://www.freereg.org.uk/csvfiles/new](https://www.freereg.org.uk/csvfiles/new)
Reported Issue:
If the server crashes when processing a file, then it loads the filename into a list called "Files NOT processed".  If the transcriber then fixes the file and tries to upload again, the server claims that a file with the same name is already on the server, and that the Replace option must be used.  However, since the file was not accepted by the server, there is no way to do a Replace.  Coordinators cannot remove files from the Not Processed list, so what transcribers have to do is rename the file.  But this is not a good solution, because if the transcriber and/or Coordinator is trying to fix the file, it may take several iterations before the server will accept it, necessitating renaming the file again and again.  A better solution is to not allow files on the Not Processed list to block uploading of the new file, or else enable transcribers to do a Replace on such files.    

![Screenshot](https://www.freereg.org.uk)

**Comment by DeniseColbert** [comment]
@Vino-S to delete the file and send the uploader and Coordinator an email with info on file not processed

**Comment by suffolkroots** [comment]
This issue has been raised many, many times and has still not been sorted. CCs need the action to delete not processed files.

**Comment by AlOneill** [comment]
I can almost hear Kirk explaining why allowing transcribers or coords to delete such files is not a good idea — I wish I could actually hear his explanation as I'm sure it's not anything trivial.

**Comment by edickens** [comment]
Hi Ali,

The problem is that if the File is still in the queue waiting to be processed, then just adding the same file again to the queue will mean that the new file is NOT an amendment of the one waiting in the queue, but the one before.  So changes will not match and the file can get into a muddle.

It would perhaps not matter if the change in the second file was just a typing error, but the server is not to know that.  There could be entries added or deleted.

To me the question is "Why is the file still waiting to be processed".  Generally the queue is very short or even non existent.  So there should not be a need to upload again before the last version was processed.  The delay is often in the transcriber receiving the confirmation email.

Patience is needed!!!

Cheers
Eric

**Comment by suffolkroots** [comment]
It should of course not be necessary for CCs to have to delete these stuck not-processed files.
The reason why they are stuck should perhaps be addressed.
This is not related to lack of patience

**Comment by suffolkroots** [comment]
ref  #1511  and #1527

The problem files often have crashed because of "illegal quoting".

IMHO, the processing of such files should be allowed, with any "illegal quotings" accepted as errors and all of them specified on the email report, in order that they may be corrected and the file replaced.

**Comment by DeniseColbert** [comment]
@Vino-S please could you look into the code and suggest why it might not be a good idea for CCs to be able to delete these files? Unless there is a good reason we're not aware of, it seems reasonable that Coords should be able to delete stuck/not processed files FOR THEIR COUNTIES. They already have the ability to delete county files: why is this different? 


~~Dale suggests we intercept the error message and send a message with wtte: Your file could not be processed. This could be due to a problem with the header, or the file might be in the incorrect format. (And/or perhaps put this in the help)~~ (No longer needed, fix will supersede)

**Comment by DeniseColbert** [comment]
Vino is working to have the problem files auto-deleted

**Comment by DeniseColbert** [comment]
In Test 3, @stsccfr to test

**Comment by stsccfr** [comment]
This still isn't working on test3.  I uploaded a junk file called STSABCBA.CSV and the test3 server rejected it and placed it onto the Files NOT Processed list.  Then I tried to upload a legitimate transcription file with the same name.  The server immediately returned the error "You cannot upload a file with the same name as one you have already uploaded. You must use the REPLACE option."

**Comment by DeniseColbert** [comment]
@Vino-S  to take a look

**Comment by DeniseColbert** [comment]
When deployed to live, @Vino-S will let Dale know so he can test

**Comment by Vino-S** [comment]
Deployed to production. 
Batch gets deleted when unprocessed.

**Comment by Vino-S** [comment]
@stsccfr could you test it on live, please?

**Comment by stsccfr** [comment]
I just tested this on live and it is still not working.  I logged in as an ordinary transcriber and uploaded a bad file that was rejected by the server with a parse error, then I tried to upload a fixed version of the same file, but I still get the message that there is already a file of that name on the server.  Then I logged in as Syndicate Coordinator and checked the Files Not Processed list, and I see the file there.  So rejected files are still being put on that list somehow.

**Comment by Vino-S** [comment]
Hi @stsccfr ,
I have updated the code. Could you give a test again, please?

**Comment by stsccfr** [comment]
Yes, it is working now.  After receiving the rejection email for the bad file, the system allowed me to upload the corrected file as a new file, and it was accepted by the server.
